public/components/health-check/container/health-check.container.tsx

```diff
@@ -0,0 +1,187 @@
+ *
+ */
+
+import {
+  EuiButton,
+  EuiCallOut,
+  EuiDescriptionList,
+  EuiSpacer,
+} from '@elastic/eui';
+import React, { Fragment, useState, useEffect, useRef } from 'react';
+import { AppState, ErrorHandler } from '../../../react-services';
+import { useAppConfig } from '../../../components/common/hooks';
+import {
+  checkPatternService,
+  checkTemplateService,
+  checkApiService,
+  checkSetupService,
+  checkFieldsService,
+  checkKibanaSettings,
+  checkKibanaSettingsTimeFilter,
+  checkPatternSupportService
+} from '../services';
+import { CheckResult } from '../components/check-result';
+import { withReduxProvider } from '../../common/hocs';
+import { getHttp } from '../../../kibana-services';
+import { WAZUH_INDEX_TYPE_MONITORING, WAZUH_INDEX_TYPE_STATISTICS } from '../../../../common/constants';
+
+const checks = {
+  api: {    
+    title: 'Check Wazuh API connection',
+    validator: checkApiService,
+    awaitFor: [],
+    canRetry: true,
+  },
+  setup: {
+    title: 'Check Wazuh API version',
+    validator: checkSetupService,
+    awaitFor: ["api"],
+  },
+  pattern: {
+    title: 'Check alerts index pattern',
+    validator: checkPatternService,
+    awaitFor: [],
+    canRetry: true,
+  },
+  template: {
+    title: 'Check alerts indices template',
+    validator: checkTemplateService,
+    awaitFor: ["pattern"],
+    canRetry: true,
+  },
+  fields: {
+    title: 'Check alerts index pattern fields',
+    validator: checkFieldsService,
+    awaitFor: ["pattern"],
+    canRetry: true,
+  },
+  patternMonitoring: {
+    title: 'Check monitoring index pattern',
+    validator: (appConfig) => checkPatternSupportService(appConfig.data['wazuh.monitoring.pattern'], WAZUH_INDEX_TYPE_MONITORING),
+    awaitFor: [],
+    shouldCheck: true,
+    canRetry: true,
+  },
+  patternStatistics: {
+    title: 'Check statistics index pattern',
+    validator: (appConfig) => checkPatternSupportService(`${appConfig.data['cron.prefix']}-${appConfig.data['cron.statistics.index.name']}-*`, WAZUH_INDEX_TYPE_STATISTICS),
+    awaitFor: [],
+    shouldCheck: true,
+    canRetry: true,
+  }
+};
+
+function HealthCheckComponent() {
+  const [checkErrors, setCheckErrors] = useState<{[key:string]: []}>({});
+  const [checksReady, setChecksReady] = useState<{[key: string]: boolean}>({});
+  const appConfig = useAppConfig();
+  const checksInitiated = useRef(false);
+
+  useEffect(() => {
+    if (appConfig.isReady && !checksInitiated.current) {
+      checksInitiated.current = true;
+      checkKibanaSettings(appConfig.data['checks.metaFields']);
+      checkKibanaSettingsTimeFilter(appConfig.data['checks.timeFilter']);
+      AppState.setPatternSelector(appConfig.data['ip.selector']);
+    }
+  }, [appConfig]);
+
+  useEffect(() => {
+    // Redirect to app when all checks are ready
+    Object.keys(checks)
+      .every(check => checksReady[check]) && (window.location.href = getHttp().basePath.prepend('/app/wazuh#/overview'));
```

Comment:
when the health check is done and ok, it maybe would return to the prev location, but it would go to overview if there is the first navigation in the app.

Reply:
👍